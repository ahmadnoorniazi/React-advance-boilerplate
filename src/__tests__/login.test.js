// these should normally be in your jest setupTestFrameworkScriptFile
import "jest-dom/extend-expect";
import "react-testing-library/cleanup-after-each";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { fireEvent, cleanup, render } from "react-testing-library";

import reducer from "../reducers";
import React from "react";
import LoginForm from "../Login";

afterEach(cleanup);
test("should test for login ", () => {
  const store = createStore(reducer);
  window.HTMLElement.prototype.scrollIntoView = function() {};
  window.alert = function() {};
  const fakeHistory = { push: jest.fn() };
  const { container, getByText, getByLabelText } = render(
    <Provider store={store}>
      <LoginForm history={fakeHistory} />
    </Provider>
  );
  const userEmail = getByLabelText("Email");
  const userPassword = getByLabelText("Password");
  const SubmitNode = getByText("Submit");

  fireEvent.change(userEmail, { target: { value: "admin" } });
  fireEvent.change(userPassword, { target: { value: "admin" } });
  fireEvent.submit(SubmitNode);
  expect(SubmitNode.type).toBe("submit");
  expect(fakeHistory.push).toHaveBeenCalledTimes(1);
  expect(fakeHistory.push).toHaveBeenCalledWith("./chat");
  expect(container.firstChild).toMatchSnapshot();
});
