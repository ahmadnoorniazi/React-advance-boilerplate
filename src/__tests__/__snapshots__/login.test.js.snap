// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test for login  1`] = `
<div
  class="form-box"
>
  <div
    class="head"
    name="loginForm"
  >
    Login Form
  </div>
  <form>
    <div
      class="form-group"
    >
      <label
        class="label-control"
        for="email"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        placeholder="User Name"
        type="text"
        value="admin"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="label-control"
        for="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        placeholder="password"
        type="password"
        value="admin"
      />
    </div>
    <button
      class="submit"
      name="Submit"
      type="submit"
      value="Submit"
    >
      Submit
    </button>
  </form>
</div>
`;
